src/react/HomePage.js

```diff
@@ -29,185 +29,136 @@ import {
+  let flowStatus = null;
+  if (astarte.config.enableFlowPreview) {
+    [ flowStatus ] = useAstarteAPI(astarte.getFlowHealth);
   }
 
-  updateTriggerNames(response) {
-    this.setState({
-      triggers: response.data
-    });
-  }
+  const refreshData = () => {
+    console.log("refreshing data...");
+  };
 
-  handleSectionError(section, err) {
-    this.setState((state) => {
-      return (state[section] === "loading" ? {[section]: "err"} : state)
+  useEffect(() => {
+    astarte.addListener('credentialsChange', refreshData);
+    return (() => {
+      astarte.removeListener('credentialsChange', refreshData);
     });
-  }
+  }, [astarte]);
 
-  updateStatus(service, status) {
-    this.setState({
-      [service]: status
-    });
-  }
-
-  redirectToLastInterface(e, interfaceName) {
+  const redirectToLastInterface = (e, interfaceName) => {
     e.preventDefault();
 
-    const reactHistory = this.props.history;
-
-    this.astarte.getInterfaceMajors(interfaceName)
+    astarte.getInterfaceMajors(interfaceName)
       .then((response) => {
         const latestMajor = Math.max(...response.data);
-        reactHistory.push(`/interfaces/${interfaceName}/${latestMajor}`);
+        history.push(`/interfaces/${interfaceName}/${latestMajor}`);
       });
-  }
+  };
 
-  render() {
-    const cellSpacingClass = "mb-3";
-
-    const {
-      connectedDevices,
-      totalDevices,
-      appengineStatus,
-      realmManagementStatus,
-      pairingStatus,
-      flowStatus,
-      interfaces,
-      triggers
-    } = this.state;
-
-    return (
-      <Container fluid className="p-3">
-        <Row>
-          <Col xs={12}>
-            <h2 className="mb-4">Astarte Dashboard</h2>
+  const cellSpacingClass = "mb-3";
+
+  return (
+    <Container fluid className="p-3">
+      <Row>
+        <Col xs={12}>
+          <h2 className="mb-4">Astarte Dashboard</h2>
+        </Col>
+        <Col xs={6} className={cellSpacingClass}>
+          <ApiStatusCard
+            appengine={appengineStatus}
+            realmManagement={realmManagementStatus}
+            pairing={pairingStatus}
+            flow={flowStatus}
+          />
+        </Col>
+        <WaitForData status={deviceStatusStats}>
+          <Col xs={6} className={cellSpacingClass}>
+            <DevicesCard
+              connectedDevices={devicesStats.connected_devices}
+              totalDevices={devicesStats.total_devices}
+            />
           </Col>
+        </WaitForData>
+        <WaitForData status={interfacesStats}>
           <Col xs={6} className={cellSpacingClass}>
-            <ApiStatusCard
-              appengine={appengineStatus}
-              realmManagement={realmManagementStatus}
-              pairing={pairingStatus}
-              flow={flowStatus}
+            <InterfacesCard
+              interfaceList={interfaces}
+              onInterfaceClick={redirectToLastInterface}
+              onInstallInterfaceClick={() => { history.push("/interfaces/new") }}
```

Comment:
You can omit parentheses for the function body: `() => history.push("/interfaces/new")`

Reply:
Omitting the parentheses will return the actual value returned by `push`.
It doesn't make a difference here, but it empathize that i don't expect to use any return value from that function